packages/account/src/providers/transaction-summary/operations.ts

```diff
@@ -252,7 +257,12 @@ export function getContractCallOperations({
     if (contractInput) {
       const newCallOps = contractCallReceipts.reduce((prevContractCallOps, receipt) => {
         if (receipt.to === contractInput.contractID) {
-          const input = getInputFromAssetId(inputs, receipt.assetId);
+          let input;
+          if (receipt.assetId !== ZeroBytes32) {
+            input = getRelevantInputs(inputs, receipt.assetId);
+          } else {
+            input = getRelevantInputs(inputs);
+          }
```

Comment:
2. The `assetId` property for the call receipt will always be `0x0000...` unless amounts are being forwarded to the contract call.

This part of the code tries to identify the account address responsible for the contract call by checking the first input with the same `assetId` as the amounts forwarded to the contract call.

However, if no amount is being forwarded and the `assetId` is a zeroed hex, we need to look for the first input with the `assetId` of the base asset.

So we can change this to:
```ts
if (receipt.to === contractInput.contractID) {
  const assetId = receipt.assetId === ZeroBytes32 ? baseAssetId : receipt.assetId;
  const input = getInputFromAssetId(inputs, assetId, baseAssetId);
```

Reply:
> 2. The assetId property for the call receipt will always be 0x0000... unless amounts are being forwarded to the contract call.

Right, I had alluded to this [here](https://github.com/FuelLabs/fuels-ts/pull/2782#discussion_r1690388439) but thanks for the clarification. I see [this issue](https://github.com/FuelLabs/fuel-core/issues/1941) has been opened to provide a more accurate response for such a scenario. In the interim I'll use this suggestion.